pandas/core/ops.py

```diff
@@ -598,7 +598,10 @@ def _is_offset(self, arr_or_obj):
         if isinstance(arr_or_obj, pd.DateOffset):
             return True
         elif is_list_like(arr_or_obj):
-            return all(isinstance(x, pd.DateOffset) for x in arr_or_obj)
+            if len(arr_or_obj):
+                return all(isinstance(x, pd.DateOffset) for x in arr_or_obj)
```

Comment:
you can just collapse this (meaning remove the ``else`` clauses and just ``return False`` at the end of the function